src/app/page.tsx

```diff
@@ -1,101 +1,172 @@
-import Image from "next/image";
+"use client";
+import React, { useState } from "react";
+import { Box, Button, Grid2, ThemeProvider, useTheme, Tooltip } from "@mui/material";
+import FileUploaded from "@/classe/File";
+import Dropzone from "@/components/Dropzone";
+import FileList from "@/components/FileList";
+import type { DropzoneState, ImageLoadEvent, ParentDimensions } from "@/types";
 
-export default function Home() {
-  return (
-    <div className="grid grid-rows-[20px_1fr_20px] items-center justify-items-center min-h-screen p-8 pb-20 gap-16 sm:p-20 font-[family-name:var(--font-geist-sans)]">
-      <main className="flex flex-col gap-8 row-start-2 items-center sm:items-start">
-        <Image
-          className="dark:invert"
-          src="https://nextjs.org/icons/next.svg"
-          alt="Next.js logo"
-          width={180}
-          height={38}
-          priority
-        />
-        <ol className="list-inside list-decimal text-sm text-center sm:text-left font-[family-name:var(--font-geist-mono)]">
-          <li className="mb-2">
-            Get started by editing{" "}
-            <code className="bg-black/[.05] dark:bg-white/[.06] px-1 py-0.5 rounded font-semibold">
-              src/app/page.tsx
-            </code>
-            .
-          </li>
-          <li>Save and see your changes instantly.</li>
-        </ol>
+function HomePage() {
+  const theme = useTheme();
+  const [dropzoneState, setDropzoneState] = useState<DropzoneState>({
+    visible: false,
+    image_url: null,
+    fillPercentage: 0,
+  });
+  const [uploadedFiles, setUploadedFiles] = useState<FileUploaded[]>([]);
+
+  function handleSetDropzoneState(show: boolean, image_url: string | null) {
+    setDropzoneState({ visible: show, image_url });
+  }
```

Comment:
Why do we wrap setDropzoneState in another function?

Reply:
done it was a mistake from myself when i make some change